src/components/Asset/AssetActions/Trade/Swap.tsx

```diff
@@ -73,88 +75,78 @@ export default function Swap({
-          await poolInstance.getAmountOutExactIn(
-            asset.accessDetails?.addressOrId,
-            poolInfo.baseTokenAddress,
-            poolInfo.datatokenAddress,
-            amountBaseToken.toString(),
-            appConfig.consumeMarketPoolSwapFee
-          )
-        const maximumDt =
-          values.type === 'buy'
-            ? amountDataToken.greaterThan(new Decimal(maxBuyDt.tokenAmount))
-              ? maxBuyDt.tokenAmount
-              : amountDataToken.toDecimalPlaces(MAX_DECIMALS).toString()
-            : amountDataToken.greaterThan(new Decimal(balance.datatoken))
-            ? balance.datatoken
-            : amountDataToken.toDecimalPlaces(MAX_DECIMALS).toString()
-
-        const maximumBaseToken =
-          values.type === 'sell'
-            ? amountBaseToken.greaterThan(
-                new Decimal(maxBuyBaseToken.tokenAmount)
-              )
-              ? maxBuyBaseToken.tokenAmount
-              : amountBaseToken.toDecimalPlaces(MAX_DECIMALS).toString()
-            : amountBaseToken.greaterThan(new Decimal(balance.baseToken))
-            ? balance.baseToken
-            : amountBaseToken.toDecimalPlaces(MAX_DECIMALS).toString()
-
+      const datatokenLiquidity = await poolInstance.getReserve(
+        poolData.id,
+        poolData.datatoken.address
+      )
+      const baseTokenLiquidity = await poolInstance.getReserve(
+        poolData.id,
+        poolData.baseToken.address
+      )
+      if (values.type === 'buy') {
+        const maxBaseTokenFromPool = calcMaxExactIn(baseTokenLiquidity)
+
+        const maxBaseTokens = maxBaseTokenFromPool.greaterThan(
+          new Decimal(balance.baseToken)
+        )
+          ? balance.baseToken
+          : maxBaseTokenFromPool.toDecimalPlaces(MAX_DECIMALS).toString()
+
+        const maxDt = await poolInstance.getAmountOutExactIn(
+          asset.accessDetails?.addressOrId,
+          poolInfo.baseTokenAddress,
+          poolInfo.datatokenAddress,
+          maxBaseTokens.toString(),
+          appConfig.consumeMarketPoolSwapFee
+        )
+        const maximumDt = new Decimal(maxDt.tokenAmount)
+          .toDecimalPlaces(MAX_DECIMALS)
+          .toString()
         setMaximumDt(maximumDt)
-        setMaximumBaseToken(maximumBaseToken)
+        setMaximumBaseToken(maxBaseTokens)
 
         setBaseTokenItem((prevState) => ({
           ...prevState,
-          amount: amountBaseToken.toDecimalPlaces(MAX_DECIMALS).toString(),
-          maxAmount: maximumBaseToken
+          amount: balance.baseToken,
+          maxAmount: maxBaseTokens
         }))
 
         setDtItem((prevState) => ({
           ...prevState,
-          amount: amountDataToken.toDecimalPlaces(MAX_DECIMALS).toString(),
+          amount: datatokenLiquidity,
           maxAmount: maximumDt
         }))
-      } catch (error) {
-        LoggerInstance.error(error.message)
+      } else {
+        const maxDtFromPool = calcMaxExactIn(datatokenLiquidity)
```

Comment:
this looks like almost everything in this `else` statement is doing exactly the same as in above `if (values.type === 'buy')` clause, just with different const names. So CodeClimate warning for this is valid

Reply:
No, there are multiple differences. I prefer it this way that having 3-4 chained conditional operators like we had before and nobody understood anything, most of the conditions were wrong.